plugins/builds.js

```diff
@@ -8,88 +8,76 @@ const SCHEMA_ARTIFACT_ID = joi.string().label('Artifact ID');
+        server.route([{
+            method: 'GET',
+            path: '/builds/{id}/{artifact*}',
+            async handler(request, h) {
                 const buildId = request.params.id;
                 const artifact = request.params.artifact;
                 const id = `${buildId}-${artifact}`;
 
-                cache.get(id, (err, value) => {
-                    if (err) {
-                        return reply(err);
-                    }
-                    if (!value) {
-                        return reply(boom.notFound());
-                    }
+                let value;
+
+                try {
+                    value = await cache.get(id);
+                } catch (err) {
+                    return err;
+                }
 
-                    // @TODO put cache headers in here
-                    const response = reply(Buffer.from(value.c.data));
+                if (!value) {
+                    return boom.notFound();
+                }
 
-                    response.headers = value.h;
+                const response = h.response(Buffer.from(value.c.data));
 
-                    return response;
-                });
+                response.headers = value.h;
+
+                return response;
             },
-            validate: {
-                params: {
-                    id: SCHEMA_BUILD_ID,
-                    artifact: SCHEMA_ARTIFACT_ID
+            options: {
+                description: 'Read build artifacts',
+                notes: 'Get an artifact from a specific build',
+                tags: ['api', 'builds'],
+                auth: {
+                    strategies: ['token'],
+                    scope: ['user']
+                },
+                plugins: {
+                    'hapi-swagger': {
+                        security: [{ token: [] }]
+                    }
+                },
+                validate: {
+                    params: {
+                        id: SCHEMA_BUILD_ID,
+                        artifact: SCHEMA_ARTIFACT_ID
+                    }
                 }
             }
-        }
-    }, {
-        method: 'PUT',
-        path: '/builds/{id}/{artifact*}',
-        config: {
-            description: 'Write build artifacts',
-            notes: 'Write an artifact from a specific build',
-            tags: ['api', 'builds'],
-            payload: {
-                maxBytes: parseInt(options.maxByteSize, 10) || DEFAULT_BYTES,
-                parse: false
-            },
-            auth: {
-                strategies: ['token'],
-                scope: ['build']
-            },
-            plugins: {
-                'hapi-swagger': {
-                    security: [{ token: [] }]
-                }
-            },
-            handler: (request, reply) => {
-                const username = request.auth.credentials.username;
+        }, {
+            method: 'PUT',
+            path: '/builds/{id}/{artifact*}',
+            async handler(request, h) {
```

Comment:
what's the difference between this and `handler: async function (request, h)`? just found that in their doc https://hapijs.com/api

Reply:
Nothing 😛 

We have a linting rule, `object-shorthand`, that enforces the following rules:

```js
{
  // allowed
  async handler(request, h) {

  },

  // allowed
  handler: async (request, h) => {

  },

  // NOT allowed
  handler: async function (request, h) {

  }
}
```

Our linting rules are defined in the following repository: https://github.com/screwdriver-cd/eslint-config-screwdriver

Do you have a preference as to which style we should use?

Reply:
Nice examples! i actually like the second one, feel more clear that way